datafusion/src/physical_plan/functions.rs

```diff
@@ -543,6 +553,28 @@ macro_rules! invoke_if_unicode_expressions_feature_flag {
     };
 }
 
+///Function volatility determines when a function can be inlined or in the future have evaluations elided when the arguments are the same
+///Immutable - a pure function which always remains the same
+///Stable - Maybe not required?. Used for functions which can modify the db but are stable for multiple calls in statement
+///Volatile - Functions where output can vary for each call.
+///For more information see https://www.postgresql.org/docs/current/xfunc-volatility.html
+#[derive(Clone, Copy, PartialEq, Eq, Hash, PartialOrd, Ord)]
+#[repr(u32)]
+pub enum FunctionVolatility {
+    ///Immutable - a pure function which always remains the same
+    Immutable = 0,
+    ///Stable - Maybe not required?. Used for functions which can modify the db but are stable for multiple calls in statement is this relevant for datafusion as in mem?
+    Stable = 1,
+    ///Volatile - Functions where output can vary for each call.
+    Volatile = 2,
+}
+
+impl Default for FunctionVolatility {
```

Comment:
IMO there is no default volatility (i.e. I am -1 on this). Defaults make it very easy for someone to just use the default and ignore the semantics of volatility.

For example, the volatility in spark never made it to pyspark, which made it impossible for users to write immutable UDFs, which forced some filtering pushdown to not be applied (even if the user _knows_ that the UDF is immutable).

Reply:
That makes a lot of sense. I'll remove the default volatility.